### ISSUE TITLE
REPL Tab Completion on Unicode Filenames

### ISSUE THREAD

**Issue by JonathanAnderson** [JuliaLang/julia#10324]
The setup

    shell> mkdir /tmp/julia_test
    julia> for i=0:9; run(`touch /tmp/julia_test/λ.$i.test`); end

The first test:

    shell> ls /tmp/julia_test/<TAB>

correctly completes to

    shell> ls /tmp/julia_test/λ.

The second (failing) test:

    shell> ls /tmp/julia_test/λ<TAB>

incorrectly erases the final / and completes to

    shell> ls /tmp/julia_testλ.

This does not happen if the filenames are named `L.$i.test`

Finally:

```
julia> versioninfo()
Julia Version 0.3.5-pre+121
Commit e424d33 (2015-01-07 00:19 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

**Comment by jiahao** [comment]
If you go back and reinspect the line after pressing return, do you see the missing /? Some systems have incorrect character width data, causing the display at a console to be funky.

**Comment by JonathanAnderson** [comment]
If i continue pressing tab after the incorrect path shows up, there is no completion output.

If I run the command that completion generated, I get the following output

```
shell> ls /tmp/julia_testλ.
ls: cannot access /tmp/julia_testλ.: No such file or directory
```

**Comment by JeffBezanson** [comment]
Also seems to exist in 0.4.

**Comment by dhoegh** [comment]
I have a fix in #10336.
